### ISSUE TITLE
[Firefox] Instant injection mode combined with run-at document-start breaks injection

### ISSUE THREAD

**Issue by lainverse** [Tampermonkey/tampermonkey#1613]
Open https://www.wikipedia.org/
Open dev console

## Expected Behavior
Script runs, you can see 1 in the console.

## Actual Behavior
Script doesn't run and there's nothing in the console.

## Specifications

- Firefox: 105.0.3 (64-bit)
- TM: 4.18.0 (stable)
- OS: Windows 10 

## Script

```js
// ==UserScript==
// @name         New Userscript
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://www.wikipedia.org/
// @icon         https://www.google.com/s2/favicons?sz=64&domain=wikipedia.org
// @grant        none
// @run-at       document-start
// ==/UserScript==

(function() {
    'use strict';

    console.log(1);
})();
```

**Comment by derjanb** [comment]
Should be fixed at 4.19.6170 (in review|[crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|[xpi](https://firefox.tampermonkey.net/firefox-current-beta.xpi))

Please install the [BETA version](https://www.tampermonkey.net/index.php?browser=firefox) or check for BETA version updates at `about:addons` 

For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.

**Comment by dimisa-RUAdList** [comment]
I tested this version in Firefox 106.0.2. Everything seems to be in order.
